test/functional/p2p_blocksonly.py

```diff
@@ -55,8 +55,51 @@ def run_test(self):
         with self.nodes[0].assert_debug_log(['received getdata for: tx {} peer=1'.format(txid)]):
             self.nodes[0].sendrawtransaction(sigtx)
             self.nodes[0].p2p.wait_for_tx(txid)
+            pprint(self.nodes[0].p2p.wait_for_tx(txid))
             assert_equal(self.nodes[0].getmempoolinfo()['size'], 1)
 
+        self.log.info("Restarting node 0 with whitelist permission and blocksonly")
+        self.stop_node(0)
+        self.start_node(0, ["-whitelist=127.0.0.1", "-whitelistforcerelay", "-blocksonly"])
```

Comment:
There seem to exist many variations of the "whitelist" params and different tests set these params differently. For example, ` ["-whitelist=127.0.0.1", "-whitelistrelay=0"]`, or ` ["-whitelist=127.0.0.1", "-whitelistrelay=0"],`, `['-whitelist=noban@127.0.0.1']`, etc. Understanding the nuances between them is a bit challenging after I have done quite a bit of digging. Could someone point me to some resource for this?

Reply:
REVISA LA BASE Y CONTINUA CON LO DEMÁS.